src/js/node/_http_incoming.ts

```diff
@@ -341,95 +383,369 @@ const IncomingMessagePrototype = {
+      this.removeAllListeners("data");
+      const handle = this[kHandle];
+      if (handle) {
+        handle.ondata = undefined;
+      }
+      this.resume();
     }
-    return 0;
+    return;
+  }
+
+  if (!this._dumped) {
+    this._dumped = true;
+    // If there is buffered data, it may trigger 'data' events.
+    // Remove 'data' event listeners explicitly.
+    this.removeAllListeners("data");
+    this.resume();
+  }
+};
+
+function onError(self, error, cb) {
+  // This is to keep backward compatible behavior.
+  // An error is emitted only if there are listeners attached to the event.
+  if (self.listenerCount("error") === 0) {
+    cb();
+  } else {
+    cb(error);
+  }
+}
+
+// BUN: server extras
+ObjectDefineProperty(IncomingMessage.prototype, "socket", {
+  get() {
+    if (this[kBunServer]) {
+      return (this[fakeSocketSymbol] ??= new FakeSocket(this));
+    }
+    return this.__socket;
   },
-  set httpVersionMajor(value) {
-    // noop
+  set(value) {
+    if (this[kBunServer]) {
+      this[fakeSocketSymbol] = value;
+      return;
+    }
+    this.__socket = value;
+    return;
   },
-  get httpVersionMinor() {
-    const version = this.httpVersion;
-    if (version.endsWith(".1")) {
-      return 1;
+});
+
+ObjectDefineProperty(IncomingMessage.prototype, "rawTrailers", {
+  get() {
+    if (this[kBunServer]) {
+      return [];
     }
-    return 0;
+    return this.__rawTrailers;
   },
-  set httpVersionMinor(value) {
-    // noop
+  set(value) {
+    if (this[kBunServer]) {
+      return;
+    }
+    this.__rawTrailers = value;
+    return;
   },
-  get rawTrailers() {
-    return [];
+});
+
+ObjectDefineProperty(IncomingMessage.prototype, "aborted", {
+  get() {
+    if (this[kBunServer]) {
+      return this[abortedSymbol];
+    }
+    return this.__aborted;
   },
-  set rawTrailers(value) {
-    // noop
+  set(value) {
+    if (this[kBunServer]) {
+      this[abortedSymbol] = value;
+      return;
+    }
+    this.__aborted = value;
+    return;
   },
-  get trailers() {
-    return kEmptyObject;
+});
+
+ObjectDefineProperty(IncomingMessage.prototype, "statusCode", {
+  get() {
+    if (this[kBunServer]) {
+      return this[statusCodeSymbol];
+    }
+    return this.__statusCode;
   },
-  set trailers(value) {
-    // noop
+  set(value) {
+    if (this[kBunServer]) {
+      if (!(value in STATUS_CODES)) return;
+      this[statusCodeSymbol] = value;
+      return;
+    }
+    this.__statusCode = value;
+    return;
   },
-  setTimeout(msecs, callback) {
-    void this.take;
-    const req = this[kHandle] || this[webRequestOrResponse];
+});
 
-    if (req) {
-      setRequestTimeout(req, Math.ceil(msecs / 1000));
-      if (typeof callback === "function") this.once("timeout", callback);
+ObjectDefineProperty(IncomingMessage.prototype, "statusMessage", {
+  get() {
+    if (this[kBunServer]) {
+      return this[statusMessageSymbol];
     }
-    return this;
+    return this.__statusMessage;
```

Comment:
any ideas where `__statusMessage` is in node.js' codebase? i'm having trouble finding this property.

<img width="205" height="100" alt="image" src="https://github.com/user-attachments/assets/79634790-d885-4c11-a4e8-f8fcc1c625e4" />